source/store/AppContext.js

```diff
@@ -1,14 +1,52 @@
-import PropTypes from 'prop-types';
-import React, { useState, useEffect } from 'react';
-import env from 'react-native-config';
-import StorageService, { APP_ENV_KEY } from '../services/StorageService';
+import PropTypes from "prop-types";
+import React, { useState, useEffect } from "react";
+import env from "react-native-config";
+import StorageService, {
+  APP_ENV_KEY,
+  API_ENDPOINT,
+} from "../services/StorageService";
 
 const AppContext = React.createContext();
 
+function getEndpoint(name) {
+  return {
+    baseUrl: env[`${name}_MITTHELSINGBORG_IO`] || "",
+    apiKey: env[`${name}_MITTHELSINGBORG_IO_APIKEY`] || "",
+  };
+}
+
+function* getEnvironments() {
+  if (!env.API_ENVS) {
+    yield {
+      label: "DEVELOP",
+      value: getEndpoint("DEVELOP"),
+      key: 0,
+    };
+    return;
+  }
+  const items = env.API_ENVS.split(",");
+  for (let i = 0; i < items.length; i += 1) {
+    yield {
+      label: items[i],
+      value: getEndpoint(items[i]),
+      key: i,
+    };
+  }
+}
```

Comment:
We tend to prefer a functional approach as opposed to an imperative. Disregarding that the function is a generator, I would refactor this using `.map()`, `.reduce()` and similar 👍. Something like:

```typescript
items.map(envName => ({
    label: envName,
    value: getEndpoint(envName)
}))
```

We can also probably remove the `key` property since `label` should be unique already.

Reply:
Key is needed as this is the way to set which item is selected in the list, it will be the same as label now anyways